src/engine/QueryPlanner.cpp

```diff
@@ -1923,6 +1932,9 @@ auto QueryPlanner::createSpatialJoin(
   ColumnIndex ind = aIsSpatialJoin ? jcs[0][1] : jcs[0][0];
   const Variable& var =
       otherSubtreePlan._qet->getVariableAndInfoByColumnIndex(ind).first;
+  std::cout << spatialJoin->onlyForTestingGetTask().first << " "
+            << spatialJoin->onlyForTestingGetTask().second << " Add variable "
+            << var.name() << "\n";
```

Comment:
TODO: bug to address here

Reply:
This bug is also present in the previous implementation: 

```sparql
PREFIX geo: <http://www.opengis.net/ont/geosparql#>
SELECT * WHERE {
  ?a geo:hasCentroid/geo:asWKT ?left .
  ?left <max-distance-in-meters:500> ?right1 .
  ?b geo:hasCentroid/geo:asWKT ?centroidR .
  ?left <max-distance-in-meters:200> ?right2 .
  ?c geo:hasCentroid/geo:asWKT ?centroidRR .
}
```

gives

`SpatialJoin needs two children, but at least one is missing. In file "/home/chris/Documents/qlever/src/engine/SpatialJoin.cpp " at line 91`

It can be worked around by binding a copy of the left variable:   `BIND(?left AS ?left2)`

Reply:
To reproduce in new syntax:

```sparql
PREFIX spatial: <https://qlever.cs.uni-freiburg.de/spatialSearch/>
PREFIX geo: <http://www.opengis.net/ont/geosparql#>
SELECT * WHERE {
  ?a geo:hasCentroid/geo:asWKT ?left .
  SERVICE spatial: {
    _:config spatial:algorithm spatial:s2 ;
             spatial:left ?left ;
             spatial:right ?right ;
             spatial:maxDistance 500 ;
             spatial:bindDistance ?dist1 .
    {
      ?b geo:hasCentroid/geo:asWKT ?right1 .
    }
  }
  SERVICE spatial: {
    _:config spatial:algorithm spatial:s2 ;
             spatial:left ?left ;
             spatial:right ?right2 ;
             spatial:maxDistance 200 ;
             spatial:bindDistance ?dist2 .
    {
      ?c geo:hasCentroid/geo:asWKT ?right2 .
    }
  }
}
```

Reply:
Apologies for the mistake. I renamed variables inconsistently in the example. The correct query is of course:

```sparql
PREFIX geo: <http://www.opengis.net/ont/geosparql#>
SELECT * WHERE {
  ?a geo:hasCentroid/geo:asWKT ?left .
  ?left <max-distance-in-meters:500> ?right1 .
  ?b geo:hasCentroid/geo:asWKT ?right1 .
  ?left <max-distance-in-meters:200> ?right2 .
  ?c geo:hasCentroid/geo:asWKT ?right2 .
}
```

This produces the bug exactly like I described.